dev/bots/service_worker_test.dart

```diff
@@ -842,3 +854,110 @@ Future<void> runWebServiceWorkerTestWithCustomServiceWorkerVersion({
   }
   print('END runWebServiceWorkerTestWithCustomServiceWorkerVersion(headless: $headless)');
 }
+
+/// Regression test for https://github.com/flutter/flutter/issues/111866.
+Future<void> runWebServiceWorkerTestWithCustomBaseHref({
+  required bool headless,
+}) async {
+  const String baseHref = 'sub';
+  await _generateEntrypoint();
+  final Map<String, int> requestedPathCounts = <String, int>{};
+  void expectRequestCounts(Map<String, int> expectedCounts) =>
+      _expectRequestCounts(expectedCounts, requestedPathCounts);
+
+  AppServer? server;
+  Future<void> waitForAppToLoad(Map<String, int> waitForCounts) async =>
+      _waitForAppToLoad(waitForCounts, requestedPathCounts, server);
+
+  Future<void> startAppServer({
+    required String cacheControl,
+  }) async {
+    final int serverPort = await findAvailablePortAndPossiblyCauseFlakyTests();
+    final int browserDebugPort = await findAvailablePortAndPossiblyCauseFlakyTests();
+    server = await AppServer.start(
+      headless: headless,
+      cacheControl: cacheControl,
+      // TODO(yjbanov): use a better port disambiguation strategy than trying
+      //                to guess what ports other tests use.
+      appUrl: 'http://localhost:$serverPort/$baseHref/index.html',
+      serverPort: serverPort,
+      browserDebugPort: browserDebugPort,
+      appDirectory: _appBuildDirectory,
+      additionalRequestHandlers: <Handler>[
+        (Request request) {
+          final String requestedPath = request.url.path;
+          requestedPathCounts.putIfAbsent(requestedPath, () => 0);
+          requestedPathCounts[requestedPath] = requestedPathCounts[requestedPath]! + 1;
+          if (requestedPath == '$baseHref/CLOSE') {
+            return Response.ok('OK');
+          }
+          return Response.notFound('');
+        },
+      ],
+    );
+  }
+
+  // Preserve old index.html as index_og.html so we can restore it later for other tests
+  await runCommand(
+    'mv',
+    <String>[
+      'index.html',
+      'index_og.html',
+    ],
+    workingDirectory: _testAppWebDirectory,
+  );
+
+  print('BEGIN runWebServiceWorkerTestWithCustomBaseHref(headless: $headless)');
+
+  try {
+    await _rebuildApp(
+      version: 1,
+      testType: ServiceWorkerTestType.generatedEntrypoint,
+      target: _target,
+      baseHref: '/$baseHref/',
+      output: './build/web/$baseHref'
+    );
+    await startAppServer(cacheControl: 'max-age=3600');
+    await waitForAppToLoad(<String, int>{
+      '$baseHref/CLOSE': 1,
+      '$baseHref/flutter_service_worker.js': 1,
+    });
+
+    expectRequestCounts(<String, int>{
+      '$baseHref/index.html': 2,
+      '$baseHref/flutter.js': 1,
+      '$baseHref/main.dart.js': 1,
+      '$baseHref/flutter_service_worker.js': 1,
+      '$baseHref/CLOSE': 1,
+      '$baseHref/assets/FontManifest.json': 1,
+      '$baseHref/assets/AssetManifest.json': 1,
```

Comment:
I need to update the naming here due to the recent changes. Will do it after the review

Reply:
For web, `AssetManifest.json` should be `AssetManifest.bin.json` since https://github.com/flutter/flutter/pull/131382

Reply:
Done